### PR TITLE
SuiteSparse: Fix Darwin install name of shared libraries

### PR DESCRIPTION
**Pull request by jppelteret**
Related to #19508 and https://github.com/DrTimothyAldenDavis/SuiteSparse/issues/42

### CODE REVIEW

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 1]
```diff
@@ -147,6 +148,12 @@ def install(self, spec, prefix):
         make_args.append('INSTALL=%s' % prefix)
         make('install', *make_args)
 
+    @run_after('install')
+    def fix_darwin_install(self):
+        # The shared libraries are not installed correctly on Darwin:
+        if (sys.platform == 'darwin') and ('+pic' in self.spec):
+            fix_darwin_install_name(self.spec.prefix.lib)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]:
```suggestion
        if '+pic platform=darwin' not in self.spec:
            fix_darwin_install_name(self.prefix.lib)
```
Simplified things a bit. Now the import is no longer needed.

Comment (jppelteret) [reply var/spack/repos/builtin/packages/suite-sparse/package.py]: Hi @adamjstewart! Thanks for the fix. Sorry that I hadn't seen it until now. Before I commit your suggestion and un-draft this PR, may I query why if the `not` in `if '+pic platform=darwin' not in self.spec:` should rather be omitted?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/suite-sparse/package.py]: Oh yes, the `not` should be omitted. No idea why I added that, must have misread your logic.

Comment (jppelteret) [reply var/spack/repos/builtin/packages/suite-sparse/package.py]: Great, thanks for the confirmation!

**Comment by jppelteret** [comment]
Thanks once more, I appreciate you help with this one @adamjstewart and @sethrj!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**